pkg/ui/v1alpha2/backend.go

```diff
@@ -231,33 +231,33 @@ func (k *KatibUIHandler) FetchHPJobInfo(w http.ResponseWriter, r *http.Request)
 	}
 	log.Printf("Got Trial List")
 
-	resultText += "\n"
-
 	for _, t := range trialListResp.Trials {
 
-		obsLogResp, err := c.GetObservationLog(
-			context.Background(),
-			&api_pb_v1alpha2.GetObservationLogRequest{
-				TrialName: t.Name,
-				StartTime: "",
-				EndTime:   "",
-			},
-		)
-		if err != nil {
-			log.Printf("GetObservationLog from HP job failed: %v", err)
-			http.Error(w, err.Error(), http.StatusInternalServerError)
-			return
-		}
+		if t.Status.Condition == api_pb_v1alpha2.TrialStatus_SUCCEEDED {
+			obsLogResp, err := c.GetObservationLog(
+				context.Background(),
+				&api_pb_v1alpha2.GetObservationLogRequest{
+					TrialName: t.Name,
+					StartTime: "",
+					EndTime:   "",
+				},
+			)
+			if err != nil {
+				log.Printf("GetObservationLog from HP job failed: %v", err)
+				http.Error(w, err.Error(), http.StatusInternalServerError)
+				return
+			}
 
-		trialResText := make([]string, len(metricsList)+len(paramList))
-		for _, m := range obsLogResp.ObservationLog.MetricLogs {
-			trialResText[metricsList[m.Metric.Name]] = m.Metric.Value
+			trialResText := make([]string, len(metricsList)+len(paramList))
+			for _, m := range obsLogResp.ObservationLog.MetricLogs {
```

Comment:
same metricName may exist in obsLogResp.ObservationLog.MetricLogs with different value, do you plan to make trialResText stores last value for each metricName?

Reply:
This loop will return only the latest value from `obsLogResp.ObservationLog.MetricLogs` in `trialResText`, because they have the same metric name.